src/editors/containers/TextEditor/components/SourceCodeModal/hooks.js

```diff
@@ -0,0 +1,23 @@
+import { useRef } from 'react';
+import * as module from './hooks';
+
+export const getSaveBtnProps = ({ editorRef, ref, close }) => ({
+  onClick: () => {
+    if (editorRef && editorRef.current && ref && ref.current) {
+      const content = ref.current.state.doc.toString();
+      editorRef.current.setContent(content);
+      close();
+    }
+  },
+});
+
+export const preprareSourceCodeModal = ({ editorRef, close }) => {
+  const ref = useRef();
+  const saveBtnProps = module.getSaveBtnProps({ editorRef, ref, close });
+  const value = editorRef.current?.getContent();
```

Comment:
There's an error when running on www gallery. Switching to raw editor then back to visual editor will error out on this line for `_editorRef$current.getContent is not a function`.